alex-alexhollender/homework_10/1_callback_exercise/main.js

```diff
@@ -0,0 +1,96 @@
+// This file contains functions that use Ajax to request and log data from two different web services. Start by reading over the code and making sure you understand how it works.
+// Obtain API keys for both openweathermap.org and timezonedb.com. Specify the values the appropriate API keys in the weatherData and timeData objects, then verify that the Ajax requests complete successfully.
+// Next, update the app so the Ajax functionality is in a separate function that uses a callback to run other functions on the data it receives.
+// Finally, update your code so the Ajax function itself returns a function, then use it to create a getWeather() and a getTime() function.  
+
+'use strict';
+
+var weatherData = {};
+weatherData.apikey = "c78fdc82343747fbafa21a15424b715b";
```

Comment:
This is something we didn't cover in class, but an API key should be an environment variable that is hidden away from public repositories. This is because bots scan for keys and since a key is linked to you, they can use (and abuse!) it without penalty. Since the weather API is relatively harmless and your key is not of interest to bots, exposing it is probably fine; with other APIs you'll likely want to keep it hidden.